src/dmd/dinterpret.d

```diff
@@ -7419,3 +7440,35 @@ private void setValue(VarDeclaration vd, Expression newval)
     assert((vd.storage_class & (STC.out_ | STC.ref_)) ? isCtfeReferenceValid(newval) : isCtfeValueValid(newval));
     ctfeStack.setValue(vd, newval);
 }
+
+/**
+ * Removes `HookTraceImpl` if found from `ce` and `fd`.
+ * This is done by replacing `HookTraceImpl!(T, Hook, errMsg)(..., parameters)` with `Hook(parameters)`.
+ * This function is used to aid the CTFE interception of `*Trace` hooks.
+ * Parameters:
+ *  ce = The CallExp that possible will be be replaced
+ *  fd = Fully resolve function declaration that `ce` would call
+ */
```

Comment:
This comment explains "what" this is doing, but doesn't explain "why".  Someone will inherit this code, so please explain why this logic is required.